src/main/kotlin/it/pagopa/wallet/services/CdcLockService.kt

```diff
@@ -0,0 +1,38 @@
+package it.pagopa.wallet.services
+
+import it.pagopa.wallet.config.properties.RedisJobLockPolicyConfig
+import it.pagopa.wallet.exceptions.LockNotAcquiredException
+import java.util.concurrent.TimeUnit
+import org.redisson.api.RedissonReactiveClient
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+
+@Service
+class CdcLockService(
+    @Autowired private val redissonClient: RedissonReactiveClient,
+    @Autowired private val redisJobLockPolicyConfig: RedisJobLockPolicyConfig
+) {
+    private val logger: Logger = LoggerFactory.getLogger(javaClass)
+
+    fun acquireEventLock(eventId: String): Mono<Unit> {
```

Comment:
maybe this function could return a Mono<Boolean> value with true/false for lock acquired or not and propagate exceptions in case of errors.
This way we could avoid log an error for any event for which an instance have not acquired lock.

what do you think?

Reply:
Good idea, it would be cleaner and more cost-efficient

Reply:
Done in [69707ca](https://github.com/pagopa/pagopa-payment-wallet-cdc-service/pull/23/commits/69707caf0460cd7ebe1af651d8702fe7a2bd147d)